.github/workflows/on_PR_mac_matrix.yml

```
@@ -0,0 +1,40 @@
+name: Mac Matrix on PRs
+
+on: [pull_request]
+
+jobs:
+  windows:
+    name: 'MacOS - clang, BuildType:${{matrix.build_type}}, SHARED:${{matrix.shared_libraries}}'
+    runs-on: macos-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type: [Release, Debug]
+        shared_libraries: [ON, OFF]
+        
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: install dependencies
+        run: |
+          brew install ninja
+          brew install googletest
+
+      - name: Build
+        run: |
+          mkdir build && cd build
+          cmake -GNinja -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DBUILD_SHARED_LIBS=${{matrix.shared_libraries}} -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=OFF -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DCMAKE_INSTALL_PREFIX=install -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations" -DCMAKE_CXX_STANDARD=11 ..
```

```suggestion
          cmake -GNinja -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DBUILD_SHARED_LIBS=${{matrix.shared_libraries}} -DEXIV2_ENABLE_PNG=ON -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_ENABLE_BMFF=OFF -DEXIV2_TEAM_WARNINGS_AS_ERRORS=ON -DCMAKE_INSTALL_PREFIX=install -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations" -DCMAKE_CXX_STANDARD=98 ..
```

on `0.27` it should be `c++98` right?

---

I couldn't get the build to work with conan on macOS, so I tried removing the conan step. But that meant that I had to add `brew install googletest` to the installation steps and that seems to have given me a version of gtest that will only build with C++11. I figure that it's fairly harmless to use C++11 here, because I'm just passing it on the command-line rather than modifying and of the `CMakeLists.txt` files.

Do you know the best way to build on macOS?

---

Probably fine, considering that we are just trying to run these checks.

I unfortunately also don't know much about the build config, especially for MacOS and Win. 
@piponazo was the one who set most of this up AFAIK